### PR TITLE
Refactor RAIN_LEVEL to use strong enum 

### PR DESCRIPTION
**Pull request by wdarvin**

### CODE REVIEW

File: src/openrct2/world/Climate.h [hunk 1]
```diff
@@ -38,19 +38,19 @@ enum WEATHER_EFFECT
     WEATHER_EFFECT_STORM,
 };
 
-enum RAIN_LEVEL
+enum class RainLevel_e
```
Comment (ZehMatt) [inline src/openrct2/world/Climate.h:41]: We don't suffix/prefix our enums, please remove the suffix.

Comment (wdarvin) [reply src/openrct2/world/Climate.h:41]: How should handle the conflict with WeatherState::RainLevel?

Comment (ZehMatt) [reply src/openrct2/world/Climate.h:41]: You can rename it to Level as the type already states its RainLevel

**Comment by wdarvin** [comment]
Did rebase/force push to get branch up to date because I messed that up initially.  I'm less familiar with rebasing than merging so here's hoping I didn't screw anything up, but everything should be good now.

**Pull request merged**